src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -77,14 +77,19 @@ public void completed(AbstractBuild r) {
             return;
         }
         Result result = r.getResult();
-        AbstractBuild<?, ?> previousBuild = project.getLastBuild().getPreviousBuild();
+        AbstractBuild<?, ?> previousBuild = project.getLastBuild();
+        do {
+            previousBuild = previousBuild.getPreviousBuild();
+        } while (previousBuild != null && previousBuild.getResult() == Result.ABORTED);
         Result previousResult = (previousBuild != null) ? previousBuild.getResult() : Result.SUCCESS;
         if ((result == Result.ABORTED && jobProperty.getNotifyAborted())
-                || (result == Result.FAILURE && jobProperty.getNotifyFailure())
+                || (result == Result.FAILURE
+                && (previousResult != Result.FAILURE || jobProperty.getNotifyRepeatedFailure())
+                && jobProperty.getNotifyFailure())
                 || (result == Result.NOT_BUILT && jobProperty.getNotifyNotBuilt())
                 || (result == Result.SUCCESS
-                		&& (previousResult == Result.FAILURE || previousResult == Result.UNSTABLE)
```

Comment:
you'd rather use a s/\t/     /g (4 spaces) it looks nicer... didn't know about % I thought it was just the filename as in !cat % ... always used :1,$